### PR TITLE
[Windows] Add textScaleFactor support

### PR DESCRIPTION
**Pull request by moko256**
This PR replaces hard-coded value and adds text scale factor support for Windows via reading and watching registry.
The settings_plugin_unittests will cover this.
This value can be changed from Settings / Ease of Access / Display / Make text bigger.

Fixes: flutter/flutter#103641.

No changes in `flutter/tests`.

| 100% | 150% | 
|-|-|
|100% example|150% example|

<details>
<summary>Example</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(brightness: Brightness.light, primarySwatch: Colors.red),
      darkTheme:
          ThemeData(brightness: Brightness.dark, primarySwatch: Colors.red),
      home: const Content(),
    );
  }
}

class Content extends StatelessWidget {
  const Content({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text("Query accessibility features"),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text("devicePixelRatio ${MediaQuery.of(context).devicePixelRatio}"),
            Text("textScaleFactor ${MediaQuery.of(context).textScaleFactor}"),
            Text("highContrast ${MediaQuery.of(context).highContrast}"),
            Text(
                "disableAnimations ${MediaQuery.of(context).disableAnimations}"),
          ],
        ),
      ),
    );
  }
}
```
</details>

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/windows/settings_plugin_win32.cc [hunk 1]
```diff
@@ -26,29 +30,47 @@ SettingsPluginWin32::SettingsPluginWin32(BinaryMessenger* messenger,
...
   StopWatching();
   RegCloseKey(preferred_brightness_reg_hkey_);
+  RegCloseKey(text_scale_factor_reg_hkey_);
 }
 
 void SettingsPluginWin32::StartWatching() {
   if (preferred_brightness_reg_hkey_ != NULL) {
     WatchPreferredBrightnessChanged();
   }
+  if (text_scale_factor_reg_hkey_ != NULL) {
```
Comment (jnschulze) [inline shell/platform/windows/settings_plugin_win32.cc:47]: nit: It looks like you're inconsistently using `NULL`/`nullptr`.

File: shell/platform/windows/settings_plugin_win32.h [hunk 2]
```diff
@@ -39,11 +39,15 @@ class SettingsPluginWin32 : public SettingsPlugin {
 
  private:
   void WatchPreferredBrightnessChanged();
+  void WatchTextScaleFactorChanged();
 
   HKEY preferred_brightness_reg_hkey_ = NULL;
+  HKEY text_scale_factor_reg_hkey_ = NULL;
```
Comment (jnschulze) [inline shell/platform/windows/settings_plugin_win32.h:45]: nit: Use `nullptr` for initializing both HKEYs.

**Review by jnschulze — APPROVED** [review]
Thanks! LGTM with nits.

File: shell/platform/windows/settings_plugin_win32.h [hunk 3]
```diff
@@ -39,11 +39,15 @@ class SettingsPluginWin32 : public SettingsPlugin {
...
 
   std::unique_ptr<EventWatcherWin32> preferred_brightness_changed_watcher_{
       nullptr};
+  std::unique_ptr<EventWatcherWin32> text_scale_factor_changed_watcher_{
+      nullptr};
```
Comment (jnschulze) [inline shell/platform/windows/settings_plugin_win32.h:50]: nit: There's no need to explicitly initialize `unique_ptr` ;)

File: shell/platform/windows/settings_plugin_win32.cc (hunk 1 above)
Comment (moko256) [reply shell/platform/windows/settings_plugin_win32.cc]: done

File: shell/platform/windows/settings_plugin_win32.h (hunk 2 above)
Comment (moko256) [reply shell/platform/windows/settings_plugin_win32.h]: done

File: shell/platform/windows/settings_plugin_win32.h (hunk 3 above)
Comment (moko256) [reply shell/platform/windows/settings_plugin_win32.h]: done

**Comment by moko256** [comment]
Thanks for reviewing! I replaced NULL with nullptr to make consistent.

**Pull request merged**